jira/client.py

```diff
@@ -1309,6 +1310,188 @@ def update_filter(
+        payload = {
+            "description": description,
+            "displayName": display_name,
+            "teamType": team_type,
+        }
+        if site_id is not None:
+            payload["siteId"] = site_id
+        r = self._session.post(url, data=json.dumps(payload))
+        raw_team_json: dict[str, Any] = json_loads(r)
+        return Team(self._options, self._session, raw=raw_team_json)
+
+    def get_team(self, org_id: str, team_id: str, site_id: str = None) -> Team:
+        """Get the specified team.
+
+        Args:
+            org_id (str): organization identifier
+            team_id (str): team identifier
+            site_id (Optional[str])
+
+        Returns:
+            Team
+        """
+        url = f"gateway/api/public/teams/v1/org/{org_id}/teams/{team_id}"
+        if site_id is not None:
+            url += f"?siteId={site_id}"
+        r = self._session.get(url)
+        raw_team_json: dict[str, Any] = json_loads(r)
+        return Team(self._options, self._session, raw=raw_team_json)
+
+    def remove_team(
+        self,
+        org_id: str,
+        team_id: str,
+    ):
+        """Delete the specified team.
+
+        Args:
+            org_id (str): organization identifier
+            team_id (str): team identifier
+
+        Returns:
+            bool
+        """
+        url = f"gateway/api/public/teams/v1/org/{org_id}/teams/{team_id}"
+        r = self._session.delete(url)
+        if r.status_code == 204:
+            return True
+        return False
+
+    def update_team(
+        self,
+        org_id: str,
+        team_id: str,
+        description: str,
+        displayName: str,
+    ) -> Team:
+        """Modifies the specified team with new values.
+
+        Args:
+            org_id (str): organization identifier
+            team_id (str): team identifier
+
+        Returns:
+            Team
+        """
+        url = f"gateway/api/public/teams/v1/org/{org_id}/teams/{team_id}"
+
+        headers = {"Accept": "application/json", "Content-Type": "application/json"}
+
+        payload = {}
+        if description != "":
+            payload["description"] = description
+        if displayName != "":
+            payload["displayName"] = displayName
+
+        response = self._session.request(
+            "PATCH", url, data=json.dumps(payload), headers=headers
+        )
+        raw_team_json: dict[str, Any] = json_loads(response)
+        return Team(self._options, self._session, raw=raw_team_json)
+
+    def team_members(self, team_id: str, org_id: str) -> list[str]:
+        """Return the list of account Ids corresponding to the team members.
+
+        Args:
+            team_id (str): Id of the team.
+            org_id (str): Id of the org.
+
+        Returns:
+            list[str]
+        """
+        url = f"/gateway/api/public/teams/v1/org/{org_id}/teams/{team_id}/members"
+        payload = {"first": 50}
```

Comment:
this pagination implementation should be moved into a helper method, if is is specific to the Teams API I would make it a static method of the Teams Resource

Then you would call it like:

`Teams._fetch_pages(session=session, url=url, ...)`

I would aim to match as many of the parameters to the function `JIRA._fetch_pages()` if possible

Reply:
In theory it's not specific to Teams API, but isn't exactly like `JIRA._fetch_pages()` neither as it's using different identifiers for Jira's latest API (why they didn't stick to the same variable names between API idk :roll_eyes: )

For now I simply created it as a helper function but if it it's used outside of Teams it's bound to change.